src/vmm/src/device_manager/mmio.rs

```diff
@@ -197,35 +202,38 @@ impl MMIODeviceManager {
         vm: &VmFd,
         device_id: String,
         mmio_device: MmioTransport,
-        slot: &MMIODeviceInfo,
+        device_info: &MMIODeviceInfo,
     ) -> Result<()> {
         // Our virtio devices are currently hardcoded to use a single IRQ.
         // Validate that requirement.
-        if slot.irqs.len() != 1 {
+        if device_info.irqs.len() != 1 {
             return Err(Error::InvalidInput);
         }
         let identifier;
         {
             let locked_device = mmio_device.locked_device();
             identifier = (DeviceType::Virtio(locked_device.device_type()), device_id);
             for (i, queue_evt) in locked_device.queue_events().iter().enumerate() {
-                let io_addr =
-                    IoEventAddress::Mmio(slot.addr + u64::from(devices::virtio::NOTIFY_REG_OFFSET));
+                let io_addr = IoEventAddress::Mmio(
+                    device_info.addr + u64::from(devices::virtio::NOTIFY_REG_OFFSET),
+                );
                 vm.register_ioevent(queue_evt, &io_addr, i as u32)
                     .map_err(Error::RegisterIoEvent)?;
             }
-            vm.register_irqfd(locked_device.interrupt_evt(), slot.irqs[0])
-                .map_err(Error::RegisterIrqFd)?;
```

Comment:
This is why snapshot tests fail with these changes.

On the restore from snapshot path, `allocate_irq()` is never called so the IRQ FD is never registered.